### PR TITLE
Lints `src/client`

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

Title says it all.

### Context of Change

Adding linting to xrpl.js

### Type of Change

- [x] Refactor (non-breaking change that only restructures code)

## Test Plan

CI passes. Linter doesn't have any errors for `src/client`.

### CODE REVIEW

File: src/models/methods/baseMethod.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 export interface BaseRequest {
+  [x: string]: unknown;
```
Comment (mvadari) [inline src/models/methods/baseMethod.ts:2]: Not sure if this is the right way to solve the problem, but this fixes TypeScript's complaint in `client.request` that `account` doesn't exist on all `Request` objects (even though we don't need it to)

File: src/client/backoff.ts [hunk 2]
```diff
@@ -1,42 +1,69 @@
...
+   * @returns Number of attempts.
+   */
+  public get attempts(): number {
+    return this.numAttempts;
   }
 
   /**
    * Return the backoff duration.
+   *
+   * @returns The backoff duration.
```
Comment (JST5000) [inline src/client/backoff.ts:55]: nit:
```suggestion
   * @returns The backoff duration in milliseconds.
```

File: src/client/broadcastClient.ts [hunk 3]
```diff
@@ -1,44 +1,42 @@
 import { Client, ClientOptions } from ".";
 
-class BroadcastClient extends Client {
-  ledgerVersion: number | undefined = undefined;
-  private readonly _clients: Client[];
+/**
+ *
+ */
```
Comment (JST5000) [inline src/client/broadcastClient.ts:5]: Is this supposed to be blank?

File: src/client/broadcastClient.ts [hunk 4]
```diff
@@ -1,44 +1,42 @@
...
+export default class BroadcastClient extends Client {
```
Comment (JST5000) [inline src/client/broadcastClient.ts:6]: Is this class description comment supposed to be blank?

File: src/client/broadcastClient.ts [hunk 5]
```diff
@@ -1,44 +1,42 @@
...
-    this.isConnected = function () {
-      return clients.map((client) => client.isConnected()).every(Boolean);
-    };
-
-    // synchronous methods are all passed directly to the first client instance
-    const defaultClient = clients[0];
-    const syncMethods = ["sign"];
-    syncMethods.forEach((name) => {
-      this[name] = defaultClient[name].bind(defaultClient);
-    });
```
Comment (JST5000) [inline src/client/broadcastClient.ts:41]: Why do we no longer need to pass in these methods to the first client?

Comment (mvadari) [reply src/client/broadcastClient.ts:41]: We're removing `sign` from the client anyways, with the `Signer`.

**Review by natenichols — APPROVED** [review]

**Review by mukulljangid — APPROVED** [review]

**Pull request merged**